cilium-health/cmd/root.go

```diff
@@ -174,7 +173,7 @@ func runServer() {
 
 	defer server.Shutdown()
 	if err := server.Serve(); err != nil {
-		log.WithError(err).Fatal("Failed to serve cilium-health API")
+		log.WithError(err).Error("Failed to serve cilium-health API")
```

Comment:
OK, there's one other functional change here. By using `Error()` rather than `Fatal()` here, the log message will be printed, then the deferred server shutdown should be run, which I believe should allow the unix socket to be properly cleaned up when there's an error case (meaning we shouldn't need to unconditionally delete the unix socket any more, hopefully?).

Reply:
Yes, seems like that change will fix the problem with socket cleanup. Let's try to follow up with removing https://github.com/cilium/cilium/blob/master/cilium-health/launch/launcher.go#L66 after this PR gets merged.